extensions/amp-redbull-player/0.1/amp-redbull-player.js

```diff
@@ -0,0 +1,298 @@
+    element.dispatchCustomEvent(VideoEvents.LOAD);
+  }
+
+  /**
+   * @return {!JsonObject}
+   * @private
+   */
+  getMetadata_() {
+    const {sourceUrl, canonicalUrl} = Services.documentInfoForDoc(this.element);
+
+    return dict({
+      'sourceUrl': sourceUrl,
+      'canonicalUrl': canonicalUrl,
+    });
+  }
+
+  /**
+   * Sends a command to the player through postMessage.
+   * @param {string} command
+   * @private
+   */
+  sendCommand_(command) {
+    if (this.iframe_ && this.iframe_.contentWindow) {
+      if (!command.id) {
+        command.id = `${TAG}-${this.tagId_}`;
+      }
+      this.postMessage_(command);
+    }
+  }
+
+  /** @override */
+  unlayoutCallback() {
+    this.removeIframe_();
+    return true; // layout again.
+  }
+
+  /** @private */
+  removeIframe_() {
+    if (this.iframe_) {
+      removeElement(this.iframe_);
+      this.iframe_ = null;
+    }
+    if (this.unlistenFrame_) {
+      this.unlistenFrame_();
+      this.unlistenFrame_ = null;
+    }
+  }
+
+  /**
+   * @param {!Event} event
+   * @private
+   */
+  originMatches_(event) {
+    return originMatches(event, this.iframe_, /.*/);
+  }
+
+  /**
+   * @param {!Event} event
+   * @private
+   */
+  onMessage_(event) {
+    if (!this.iframe_) {
+      return;
+    }
+
+    if (!this.originMatches_(event)) {
+      return;
+    }
+
+    const eventData = getData(event);
+    if (!isJsonOrObj(eventData)) {
+      return;
+    }
+
+    const data = objOrParseJson(eventData);
+
+    if (data.id === `redbull-amp-video-tracking-${this.tagId_}`) {
+      const type = ANALYTICS_EVENT_TYPE_PREFIX + data.type;
+      this.dispatchCustomAnalyticsEvent_(type, data);
+    }
+  }
+
+  /**
+   * @param {string} eventType
+   * @param {!Object<string, string>=} vars
+   */
+  dispatchCustomAnalyticsEvent_(eventType, vars = {}) {
+    userAssert(
+      getAnalyticsEventTypePrefixRegex().test(eventType),
+      'Invalid analytics `eventType`. Value must start with `%s`.',
+      ANALYTICS_EVENT_TYPE_PREFIX
+    );
+    this.element.dispatchCustomEvent(
+      VideoEvents.CUSTOM_TICK,
+      dict({
+        'eventType': `video-custom-tracking-${this.tagId_}`,
+        'vars': vars,
+      })
+    );
+  }
+
+  /**
+   * @param {!JsonObject} message
+   * @private
+   */
+  postMessage_(message) {
+    if (!message.msg) {
+      message = JSON.stringify(message);
+    }
+
+    this.iframe_.contentWindow.postMessage(message, '*');
+  }
+  /** @override */
+  isInteractive() {
+    return true;
+  }
+
+  /** @override */
+  supportsPlatform() {
+    return true;
+  }
+
+  /** @override */
+  preimplementsMediaSessionAPI() {
+    return this.element.hasAttribute('implements-media-session');
+  }
+
+  /** @override */
+  preimplementsAutoFullscreen() {
+    return this.element.hasAttribute('implements-rotate-to-fullscreen');
+  }
+
+  /** @override */
+  getCurrentTime() {
+    return 0;
+  }
+
+  /** @override */
+  getDuration() {
+    return 0;
```

Comment:
The codebase is very inconsistent with this, but could you switch this to `NaN`?

Also for unimplemented methods, please add a comment about it being unimplemented for each.

Reply:
NaN it is